src/site/asn1.rst

```diff
@@ -68,6 +68,43 @@ country ``name`` (string), ``population`` (int), ``area`` (double), and ``contin
 Part 2 --- Country Catalogue
 ============================
 
+The ``CountryCatalogue`` class will be a collection of ``Country`` objects. There will be three fields
+
+    * ``size``, an int to keep track of the number of countries in the catalogue
+    * ``catalogue``, an array of ``Country`` objects
+    * ``countryContinent``, a *hash map* to keep track of the country/continent relationships
+        * A map is the same idea as a dictionary from Python
+        * `How does one use a hash map? <https://www.google.com/search?q=java+hashmap&oq=java+hashmap>`_
```

Comment:
An actual `HashMap`? You just [removed equals](https://github.com/jameshughes89/cs102/pull/411), but `HashMap#get` [requires both `hashCode` and `equals`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/HashMap.html#get(java.lang.Object)). By default, you'll be using Object's implementation, which is same-ness (`==`).

Reply:
For ease, I'll swap it out for `TreeMap` as I do not believe those are _required_

Reply:
TreeMap requires ordering of the objects, either via `Comparable` or a `Comparator`.

Reply:
And, that ordering is used to _imply_ equality, so it needs to be a good comparator.

Reply:
rip --- that's defo worse then. 

In that case, I suppose the easiest option is to add `equals` back and give them some basic `hashCode` method.